### ISSUE TITLE
Is it possible to support Fluent Assertions and xunit?

### ISSUE THREAD

**Issue by iron9light** [twenzel/Demystify.Everything#34]
Is it possible to support Fluent Assertions and xunit?

**Comment by twenzel** [comment]
Do you mean if it's possible to use those within the Demystify.Everything projects? FluentAssertions for sure. XUnit is already used.

**Comment by iron9light** [comment]
No, I mean when someone using FluentAssertions, xUnit or any other unit tests framework, make the output stack tracing info of exception Demystified.

**Comment by cocowalla** [comment]
Just wanted to +1 this - Demystify makes async stack traces soooo much easier to read when diagnosing xUnit unit test failures.

I think this might be possible by creating a customised `XunitTestInvoker` ([like this](https://github.com/SuperDotNet/Super.NET/blob/7eeaf44f7379d5338c01fb689178a988e32d4b5a/Super.Application.Hosting.xUnit/TestInvoker.cs)), and then using a `TestFrameworkDiscovererAttribute` to override the behaviour of `[Fact]` and `[Theory]` - there's an example doing that [here](https://github.com/SuperDotNet/Super.NET/blob/7eeaf44f7379d5338c01fb689178a988e32d4b5a/Super.Application.Hosting.xUnit/TestInvoker.cs) (the purpose was fo DI, but the principal is the same).

**Comment by twenzel** [comment]
@cocowalla Is upgrading to .NET Core 3.1 + an option? Because Ben Adams has fixed the async stack traces in .NET Core 3.1.

**Comment by cocowalla** [comment]
> @cocowalla Is upgrading to .NET Core 3.1 + an option? Because Ben Adams has fixed the async stack traces in .NET Core 3.1.

I'm on dotnet 6 now, but I think I was on dotnet 5 at the time of that comment - where Demystify was still an improvement over the standard async stack trace.
